### ISSUE TITLE
Add check whether spatial replicates were actually replicates of eachother

### ISSUE THREAD

**Issue by laurianvm** [emo-bon/data-quality-control-action#10]
copying from https://github.com/emo-bon/emobon-ontology/issues/5:
based on deployment date and collections date in sampling tab
these should be within a set time range for each replicate (ask Katrina what that time range is)
--> if they are return 1 else 0

**Comment by laurianvm** [comment]
@kmexter, this issue was created during a collaborative session. If you recall the details more clearly than I do, could you please update the description of this issue to clarify it?

Additionally, I’m wondering whether this type of check is necessary for quality control, or if an analysis using kgap might be sufficient instead.

**Comment by kmexter** [comment]
ARMS units that are spatial replicates need to be within a certain distance of each other, and deployed and retrieved within a certain timeframe of each other. The arms logsheets need to be mined to check that this is the case. However: we still need to go over the arms logsheets - @isanti needs to ensure that the logsheets as they are now are all OK and that we can go ahead with incorporating them into GH. At that point, we can then sit down with @marc-portier and decide what new QC tests need to be done. So no need to do anything on this issue until that point.

**Comment by cpavloud** [comment]
> We can use the lat long to calculate the distance, as long as they go down to the metre, so Id say no need to add a new column

But the lat/long are linked to the observatory, not the ARMS units (i.e., they do not appear in the "sampling" sheet).

**Comment by kmexter** [comment]
ok, can you guys do that (remembering that distance is either X (horizontal) and Y (depth) or is "line of sight" - whichever is most important to us)?

**Comment by isanti** [comment]
Hmmm... for potential new column "distance from replicate"
example
VH1, VH2, VH3 are replicate-units checking all the rules
VH1 replicates are VH2 and VH3, etc.
In one column "distance from replicate" which distance will it be? For a sample coming from VH1, we would need distance between VH1-VH2, VH1-VH3.  For a sample coming from VH2, we would need distance between VH1-VH2, VH2-VH3.  
And this is a bit too many distances.. I think it will be confusing for the station/sampling people. Will people complete it or just ignore it and then we will be constantly poking them to do it?

Could we just do it by calculating the horizontal distance among the coordinates of each unit?
OR add a check column whether they confirm (YES/NO) this unit was deployed in under 10m from the replicate units described in tab "observatory"? Then the responsibility (of saying Y/N replicate) would be at the stations.



> >and are the depths within WHAT of each other

>For this one, we should also probably add in the QC an extra check whether all the fractions from one replicate have the same >depth or not. Just to avoid potential errors.

ok, yes, then here a QC rule checking the depth among replicates is the same

> >timeframe differences should be <=2d

>That seems ok as a QC rule. It is relevant both at the deploy dates and at the collect dates

QC rule for this one.

**Comment by kmexter** [comment]
> Hmmm... for potential new column "distance from replicate" example VH1, VH2, VH3 are replicate-units checking all the rules VH1 replicates are VH2 and VH3, etc. In one column "distance from replicate" which distance will it be? For a sample coming from VH1, we would need distance between VH1-VH2, VH1-VH3. For a sample coming from VH2, we would need distance between VH1-VH2, VH2-VH3. And this is a bit too many distances.. I think it will be confusing for the station/sampling people. Will people complete it or just ignore it and then we will be constantly poking them to do it?
> 
> Could we just do it by calculating the horizontal distance among the coordinates of each unit? OR add a check column whether they confirm (YES/NO) this unit was deployed in under 10m from the replicate units described in tab "observatory"? Then the responsibility (of saying Y/N replicate) would be at the stations.
> 

Good point. I like the idea of a checkbox, yes. But then also a note box for the scenario where the answer is No. If anyone fills that in, then we will need a manual intervention at our end, but we can worry about that if and when it happens. 


> > > and are the depths within WHAT of each other
> 
> > For this one, we should also probably add in the QC an extra check whether all the fractions from one replicate have the same >depth or not. Just to avoid potential errors.
> 
> ok, yes, then here a QC rule checking the depth among replicates is the same
> 
> > > timeframe differences should be <=2d
> 
> > That seems ok as a QC rule. It is relevant both at the deploy dates and at the collect dates
> 
> QC rule for this one.

Depth and timeframe are on the sampling sheet, so we can easily do QC checks on that.

**Comment by isanti** [comment]
Not sure if this should go here, but it is for the QC rules for the ARMS.

I think we need another QC rule to check that all ARMS units are added in the "observatory" sheet.
In the "sampling" sheet, check that all ARMS units mentioned are included in the "observatory" sheet along with their details (coordinates etc.). 
The problem would be how the QC will understand which are the supposedly ARMS units added in the "sampling" sheet. This info is captured as text in "source_mat_id_orig", for example "EMO BON CPOWERA HA 20210623 20220622 SF40 (3)" where the ARMS unit is "CPOWERA". But "source_mat_id_orig" does not have a unique format and so capturing the ARMS unit check will be hard for the QC algorithm. This info (ARMS unit) is also captured as text in "sampling_event", but again I don't see this being always formatted the same.
Another idea would be to have a column in "sampling" for them to select the ARMS unit from the list given in the "observatory" sheet. This way we can check this easier. The logsheet will also give an error if something different is added.

@cpavloud and @kmexter 
1) do think it is important to check that? or should we just ignore?
2) if important and if we decide to check, how we do it? during the manual QC we will start we could apply changes in the logsheets so it would be great if we do this as well.

**Comment by kmexter** [comment]
yes, we need a new column in sampling, and that can be used to check
however, as we also discussed yesterday about adusting the obs_id and arms_unit_id from observatory, what exactly goes in that column may not be what you expect
So we should discuss that first (see [meeting notes](https://docs.google.com/document/d/1PW_E2goRreoQ3tAAckDsecGOfuvWkNakPvjfZlIm74w/edit?tab=t.0) and comments therein) but definitely need a new column in sampling sheet.

**Comment by cpavloud** [comment]
> Another idea would be to have a column in "sampling" for them to select the ARMS unit from the list given in the "observatory" sheet. 

I like this idea. 

So in the current logsheet, in the `observatory` sheet, we have 
`obs_id | ARMS_units_id | field_replicated_units | latitude | longitude`

In the `sampling` sheet, we will have 
`source_mat_id_orig | ARMS_units_id ` 
but the ARMS_units_id would be a dropdown. 

Initially I thought that we could move the 
`ARMS_units_id | field_replicated_units` 
to the `sampling` sheet 
but this way probably is better.

**Comment by kmexter** [comment]
>
> Was this ARMS unit deployed in under 10 m distance from its replicate units as described in the "observatory" sheet of this 
> metadata logsheet? Please add YES or NO accordingly. In case this ARMS unit was not deployed in under 10 m distance from its 
> replicate units or it drifted/moved during deployment or another event that designed the units as no-replicates, please add details
> and explanation under column "failure_comment"
>
 Suggest: 
Was this ARMS unit <= 10m from its partner units placed in the same sampling site (as described in the Observatory sheet)? Indicate Yes or No. If No, please explain why in the column failure_comment 

sampling_site is fine for me, over sampling_area

Definition:  I suggest
A specific location where your Autonomous Reef Monitoring Structures (ARMS) are deployed. Consider a unique site to be roughly a  circle of maximum 20m diameter, inside which units are deployed. Different sites need different names, and normally will be re-used from event to event over time.

**Comment by kmexter** [comment]
> I guess I overcompensated here. Can you please check? I am also thinking we don't need this long elaborative descriptions here and we could keep them for the Handbook?
> 
YES

> We can add this in the next version of the EMO BON Handbook too to describe the design and potential diverse schemes. The **Obs_id** definition will be updated in all the logsheets.
> 
YES

I am slightly worried about them adding the ARMS unit lat, long to the observatory because each year they will not be in exactly the same place - say one year they have 3 in a triangle and the next year 2 in a line, they are in the same site but say 5 m away from where they were before. do they update the lat long in the observatory spreadsheet? if so, then we will have the wrong lat long for ALL samples from those arms units, because each time we harvest the logsheets, we add the information from the observatory sheet to all those samples. So specific lat long only if they know that those units will not move from event to event by too much, and if they do then the unit needs a new name.....which is getting a bit complicated, no?

**Comment by isanti** [comment]
Ok, yes, I see. Then it would be best to only keep the sampling site coordinates and remove completely the ARMS unit coordinates.

**Comment by isanti** [comment]
closing this. Decisions and actions during Dec-25 meetings. See here for changes in the logsheets: https://github.com/emo-bon/observatory-profile/issues/77

Ecological replicate ARMS units are units placed in the same sampling site, in under or equal to (<=) 10 m among them, at the same depth and deployed-collected within a maximum 2-day timeframe. This term is checking the distance among replicate ARMS units.
QC rule:
to check the "deployed-collected within a maximum 2-day timeframe" -> check deployment_date and collection_date
to check the "same depth" -> check the depth
to check the distance ("same sampling site, in under or equal to (<=) 10 m among them") -> check the rep_check if Y or N
